sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,186 @@
+   * @param parentProcessSummary parent process summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @returns a number of update records.
+   */
+  async executeCASIntegrationProcess(
+    parentProcessSummary: ProcessSummary,
+    casSuppliers: CASSupplier[],
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    const summary = new ProcessSummary();
+    parentProcessSummary.children(summary);
+    try {
+      summary.info("Logging on CAS...");
+      const auth = await this.casService.casLogon();
+      if (auth.access_token) {
+        summary.info("Logon successful.");
+        suppliersUpdated = await this.requestCASAndUpdateSuppliers(
+          casSuppliers,
+          summary,
+          auth,
+        );
+      } else {
+        summary.info("Could not authenticate on CAS.");
+      }
+    } catch (error: unknown) {
+      summary.error("Unexpected error.", error);
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * For each pending CAS supplier, request supplier information to CAS API and update local table.
+   * @param casSuppliers pending CAS suppliers.
+   * @param summary log summary.
+   * @param auth CAS auth details.
+   * @returns a number of update records.
+   */
+  private async requestCASAndUpdateSuppliers(
+    casSuppliers: CASSupplier[],
+    summary: ProcessSummary,
+    auth: CASAuthDetails,
+  ) {
+    let suppliersUpdated = 0;
+    for (const casSupplier of casSuppliers) {
+      summary.info(`Requesting info for CAS supplier id ${casSupplier.id}.`);
+      let supplierResponse = null;
+      try {
+        supplierResponse = await this.casService.getSupplierInfoFromCAS(
+          auth.access_token,
+          casSupplier.student.sinValidation.sin,
+          casSupplier.student.user.lastName.toUpperCase(),
+        );
+      } catch (error: unknown) {
+        summary.error("Unexpected error while requesting supplier.", error);
+      }
+      if (supplierResponse.items.length > 0) {
+        const [supplierInfo] = supplierResponse.items;
+        summary.info("Updating CAS supplier table.");
+        try {
+          const updateResult = await this.updateCASSupplier(
+            casSupplier,
+            supplierInfo,
+            SupplierStatus.Verified,
+          );
+          if (updateResult.affected) {
+            suppliersUpdated++;
+          }
+        } catch (error: unknown) {
+          summary.error("Unexpected error.", error);
+        }
+      } else {
+        summary.info("No supplier found on CAS.");
+      }
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * Updates CAS supplier table.
+   * @param casSupplier CAS supplier to be updated.
+   * @param casSupplierResponseItem CAS supplier response item from CAS request.
+   * @param supplierStatus CAS supplier status to be updated.
+   * @returns update result.
+   */
+  async updateCASSupplier(
+    casSupplier: CASSupplier,
```

Comment:
Please do not pass the entire object if only the ID would be required for processing.